src/lib/protocol/README.md

```diff
@@ -741,14 +738,9 @@ pushed headers).
 — Method **datagram:payload** *pointer*, *length*
 
 Combined payload accessor and setter method. Returns a pointer to the
-datagram payload, its byte size and the `packet.iovec` index of the next
-buffer if the payload is incomplete or `nil`.
-
-The payload is incomplete if the underlying packet consists of more than
-one buffer. The caller has to access the remainder of the payload,
-e.g. any additional packet buffers, by his own.
+datagram payload and its byte size.
 
 If *pointer* and *length* are supplied then *length* bytes starting from
-*pointer* are appended to the datagram's payload. If there is not enough
-free space in the buffer to append *length* bytes or if the payload is
-incomplete an error will be raised.
+*pointer* are appended to the datagram's payload. The caller has to
```

Comment:
I think that we can remove this sentence. The app developer should not have to think about how much space is available in the packet buffer, it is the job of the core developers to ensure that this is always sufficient for any realistic scenario.

(Like we don't ask programmers to check that they have enough stack space available before calling a subroutine and instead expect the runtime system to ensure that this is the case for any sane usage.)